From: NeilBrown <neilb@suse.de>
Subject: dm-mpath: Fix parsing of "no_partitions" flag.
Patch-mainline: not-relevant
Reference: FATE#313903

SLES allows a "no_partitions" flags for dm-mpath.
Recent patch updates how flags are parsed and didn't fix "no_partitions"
(as mainline doesn't have it).  So this patch fixes it.

Signed-off-by: NeilBrown <neilb@suse.de>
Acked-by: NeilBrown <nfbrown@suse.com>

---
 drivers/md/dm-mpath.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- linux-3.0-SLE11-SP3.orig/drivers/md/dm-mpath.c
+++ linux-3.0-SLE11-SP3/drivers/md/dm-mpath.c
@@ -553,7 +553,7 @@ static struct pgpath *parse_path(struct
 {
 	int r;
 	struct pgpath *p;
-	char *path;
+	const char *path;
 	struct multipath *m = ti->private;
 
 	/* we need at least a path arg */
@@ -802,7 +802,7 @@ static int parse_features(struct dm_arg_
 			continue;
 		}
 
-		if (!strnicmp(param_name, MESG_STR("no_partitions"))) {
+		if (!strcasecmp(arg_name, "no_partitions")) {
 			m->features |= FEATURE_NO_PARTITIONS;
 			continue;
 		}
